hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/Eventloop.java

```diff
@@ -379,56 +388,67 @@ public final boolean offer(Runnable task) {
     public final boolean offer(IOBuffer buff) {
         //todo: Don't want to add localRunQueue optimization like the offer(Runnable)?
 
-        if (concurrentRunQueue.offer(buff)) {
+        if (concurrentTaskQueue.offer(buff)) {
             wakeup();
             return true;
         } else {
             return false;
         }
     }
 
-    protected final void runLocalTasks() {
-        for (; ; ) {
+    protected final boolean runScheduledTasks() {
+        for (int k = 0; k < batchSize; k++) {
             ScheduledTask scheduledTask = scheduledTaskQueue.peek();
             if (scheduledTask == null) {
-                break;
+                return false;
             }
 
-            if (scheduledTask.deadlineEpochNanos > epochNanos()) {
+            if (scheduledTask.deadlineNanos > unsafe.nanoClock.nanoTime()) {
                 // Task should not yet be executed.
-                earliestDeadlineEpochNanos = scheduledTask.deadlineEpochNanos;
-                break;
+                earliestDeadlineNanos = scheduledTask.deadlineNanos;
+                return false;
             }
 
             scheduledTaskQueue.poll();
-            earliestDeadlineEpochNanos = -1;
+            earliestDeadlineNanos = -1;
             try {
                 scheduledTask.run();
             } catch (Exception e) {
                 logger.warning(e);
             }
         }
 
-        for (; ; ) {
-            Runnable task = localRunQueue.poll();
-            if (task == null) {
-                break;
-            }
+        //todo:
+        return true;
```

Comment:
Why are we returning true here unconditionally? Can't we do `return !scheduledTaskQueue.isEmpty()` similar to others? Or, is the to-do for that?

Reply:
Good point. Fixed.